bin/main.rb

```diff
@@ -0,0 +1,103 @@
+require_relative '../lib/player'
+
+require_relative '../lib/game'
+
+game = Game.new
+
+board = game.game_arr
+
+def welcome_screen(game, board)
+  name = ''
+
+  puts 'Welcome to TIC-TAC-TOE'
+
+  puts "First player, What's your name?"
+
+  loop do
+    name = gets.chomp
+
+    unless game.validate_name(name)
+
+      puts "Your name cannot be an empty line or a number \n \n \n"
+
+      puts "Please enter a valid name \n"
+
+    end
+
+    break if game.validate_name(name) == true
+  end
+
+  game.player1.name = name
+
+  puts "Hi #{game.player1.name} , your key is #{game.player1.key} "
+
+  puts "Second player, What's your name?"
+
+  loop do
+    name = gets.chomp
+
+    unless game.validate_name(name)
+
+      puts "Your name cannot be an empty line or a number \n \n"
+
+      puts "Please enter a valid name \n"
+
+    end
+
+    break if game.validate_name(name) == true
+  end
+
+  game.player2.name = name
+
+  puts "Hi #{game.player2.name} , your key is #{game.player2.key} "
+
+  display_board(board)
+end
+
+# displaying the game board
+
+def display_board(board)
+  puts " #{board[0]} | #{board[1]} | #{board[2]} "
+
+  puts '-----------'
+
+  puts " #{board[3]} | #{board[4]} | #{board[5]} "
+
+  puts '-----------'
+
+  puts " #{board[6]} | #{board[7]} | #{board[8]} "
+end
```

Comment:
That method must be moved into the `Game` class.
Why? Because you set them in parameter
Don't you think that It will a good OOP implementation do call it like this
```ruby
puts game.display_board()
```
Where you will have this inside the `Game` class
```ruby
def display_board()
  " #{game_arr[0]} | #{game_arr[1]} | #{game_arr[2]} \n" +
  '----------- \n' +
  " #{game_arr[3]} | #{game_arr[4]} | #{game_arr[5]} \n"
  '----------- \n' +
  " #{game_arr[6]} | #{game_arr[7]} | #{game_arr[8]} \n"
end
```